stl/inc/ranges

```diff
@@ -7656,6 +7662,412 @@ namespace ranges {
+                    using _Decayed_func = decay_t<_Func>;
+
+                    // We do the static_asserts here, rather than in the definition of operator(), so that they
+                    // show up earlier in compiler diagnostics.
+                    static_assert(move_constructible<_Decayed_func>,
+                        "ERROR: When using std::views::zip_transform without any views, the specified callable must "
+                        "be move constructible (see [range.zip.transform.overview]/2.1.1).");
+
+                    if constexpr (regular_invocable<_Decayed_func&>) {
+                        static_assert(is_object_v<decay_t<invoke_result_t<_Decayed_func&>>>,
+                            "ERROR: When using std::views::zip_transform without any views, the specified callable "
+                            "must return an object (see [range.zip.transform.overview]/2.1.1).");
+
+                        // NOTE: As of writing this, the MSVC hasn't implemented auto(x)/auto{x}
+                        // (see P0849R8). We need to thus manually describe the type instead of using
+                        // auto(std::views::empty<std::decay_t<std::invoke_result_t<_Decayed_func&>>>)
+                        // as the C++ standard proposes.
+                        //
+                        // (The alternative is to use the exposition-only function decay-copy, but this
+                        // isn't what's written in the standard anymore.)
+                        using _Empty_view_t = decltype(empty<decay_t<invoke_result_t<_Decayed_func&>>>);
+
+                        return noexcept(((void) _STD declval<_Func&>(),
+                            _Empty_view_t{empty<decay_t<invoke_result_t<_Decayed_func&>>>}));
+                    } else {
+                        static_assert(_Always_false<_Decayed_func>,
+                            "ERROR: When using std::views::zip_transform without any views, the specified callable "
+                            "must be invocable without any parameters (see [range.zip.transform.overview]/2.1.1).");
+                        return false;
+                    }
+                } else {
+                    return noexcept(noexcept(zip_transform_view(_STD declval<_Func>(), _STD declval<_Types>()...)));
+                }
+            }();
+
+        public:
+            template <class _Func, class... _Types>
+            _NODISCARD constexpr auto operator()(_Func&& _Function, _Types&&... _Args) const
+                noexcept(_Is_invocation_noexcept<_Func, _Types...>) // strengthened
+            {
+                if constexpr (sizeof...(_Types) == 0) {
+                    using _Decayed_func = decay_t<decltype((_Function))>;
+                    using _Empty_view_t = decltype(empty<decay_t<invoke_result_t<_Decayed_func&>>>);
+
+                    return ((void) _Function, _Empty_view_t{empty<decay_t<invoke_result_t<_Decayed_func&>>>});

```

Comment:
Ditto for `empty_view<T>{}`. MSVC STL is already doing so for `views::take`.

https://github.com/microsoft/STL/blob/8ddf4da23939b5c65587ed05f783ff39b8801e0f/stl/inc/ranges#L3126-L3127

```suggestion

                    // equivalent to:
                    // return ((void) _Function, auto(empty<decay_t<invoke_result_t<_Decayed_func&>>>));
                    return empty_view<decay_t<invoke_result_t<_Decayed_func&>>>{};
```